SlackSocketClient.cs

```diff
@@ -67,12 +67,18 @@ public void SendTyping(string channelId)
             underlyingSocket.Send(new Typing() { channel = channelId });
         }
 
-        public void SendMessage(Action<MessageReceived> onSent, string channelId, string textData)
+        public void SendMessage(Action<MessageReceived> onSent, string channelId, string textData, string userName = null)
         {
-			underlyingSocket.Send(new Message() { channel = channelId, text = textData, user = MySelf.id, type = "message" }, new Action<MessageReceived>((mr) => {
-				if(onSent != null)
-					onSent(mr);
-			}));
+            if (onSent != null)
```

Comment:
Oh good grief, I'm sorry -- that was a bad merge gone wrong (I was cherry picking from the middle of my "EventArgs" change) -- the outer if isn't needed if one just tweaks it like ...